src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -0,0 +1,160 @@
+import { BasicField } from '../../../../types'
+import {
+  ProcessedCheckboxResponse,
+  ProcessedTableResponse,
+} from '../submission.types'
+
+import {
+  EmailAutoReplyField,
+  EmailDataForOneField,
+  EmailJsonField,
+  ResponseFormattedForEmail,
+} from './email-submission.types'
+
+/**
+ * Determines the prefix for a question based on whether it is verified
+ * by MyInfo.
+ * @param {Object} response
+ * @param {Set} hashedFields Hash for verifying MyInfo fields
+ * @returns {String}
+ */
+const getMyInfoPrefix = (
+  response: ResponseFormattedForEmail,
+  hashedFields: Set<string>,
+): string => {
+  return !!response.myInfo?.attr && hashedFields.has(response._id)
+    ? '[MyInfo] '
+    : ''
+}
+
+/**
+ * Determines the prefix for a question based on whether it was verified
+ * by a user during form submission.
+ * @param response
+ * @returns the prefix
+ */
+const getVerifiedPrefix = (response: ResponseFormattedForEmail): string => {
+  return response.isUserVerified ? `[verified] ` : ''
+}
+
+/**
+ * Determines the prefix for a question based on its field type.
+ * @param fieldType
+ * @returns the prefix
+ */
+const getFieldTypePrefix = (response: ResponseFormattedForEmail): string => {
+  switch (response.fieldType) {
+    case BasicField.Table:
+      return `[table] `
+    case BasicField.Attachment:
+      return `[attachment] `
+    default:
+      return ''
+  }
+}
+
+/**
+ * Transforms a question for inclusion in the JSON data used by the
+ * data collation tool.
+ * @param response
+ */
+export const getJsonPrefixedQuestion = (
+  response: ResponseFormattedForEmail,
+): string => {
+  const questionComponents = [getFieldTypePrefix(response), response.question]
+  return questionComponents.join('')
+}
+
+/**
+ * Transforms a question for inclusion in the admin email table.
+ * @param response
+ * @param hashedFields
+ */
+export const getFormDataPrefixedQuestion = (
+  response: ResponseFormattedForEmail,
+  hashedFields: Set<string>,
+): string => {
+  const questionComponents = [
+    getFieldTypePrefix(response),
+    getMyInfoPrefix(response, hashedFields),
+    getVerifiedPrefix(response),
+    response.question,
+  ]
+  return questionComponents.join('')
+}
+
+/**
+ * Creates one response for every row of the table using the answerArray
+ * @param response
+ * @param response.answerArray an array of array<string> for each row of the table
+ * @returns array of duplicated response for each answer in the answerArray
+ */
+export const getAnswerRowsForTable = (
+  response: ProcessedTableResponse,
+): ResponseFormattedForEmail[] => {
+  return response.answerArray.map((answer) => {
+    return Object.assign({}, response, { answer: String(answer) })
```

Comment:
This still contains the `answerArray` key leh, should we omit the key first?

```ts
return response.answerArray.map((answer) => {
  const { answerKey, ...omitAnswerArrayObj } = response
  return {
    ...omitAnswerArrayObj,
    answer: String(answer)
  }
})
```
(or we can use `_.omit` lul)

nit: also not a fan of Object.assign leh, not very obvious at first glance what is happening

Reply:
yeah, I'm not a big fan either but I just kept the existing code. scared things will break if we start omitting keys. but I can try...